### PR TITLE
[vcpkg-scripts] Fix rpath fixup; add test

### PR DESCRIPTION
**Pull request by dg0yt**
Fix fixup for debug tools, https://github.com/microsoft/vcpkg/pull/37736#issuecomment-2035415913.  
Fix fixup for paths with regex chars, https://github.com/microsoft/vcpkg/issues/37984.

### CODE REVIEW

File: scripts/cmake/z_vcpkg_fixup_rpath.cmake [hunk 1]
```diff
@@ -9,7 +9,8 @@ function(z_vcpkg_calculate_corrected_rpath)
 
     set(current_prefix "${CURRENT_PACKAGES_DIR}")
     set(current_installed_prefix "${CURRENT_INSTALLED_DIR}")
-    if(elf_file_dir MATCHES "debug/")
+    file(RELATIVE_PATH relative_from_packages "${CURRENT_PACKAGES_DIR}" "${elf_file_dir}")
+    if("${relative_from_packages}/" MATCHES "^debug/|^(manual-tools|tools)/[^/]*/debug/")
```
Comment (Neumann-A) [inline scripts/cmake/z_vcpkg_fixup_rpath.cmake:13]:
```suggestion
    if("${relative_from_packages}/" MATCHES "^debug/|^((manual-)?tools)/[^/]*/debug/")
```

Comment (dg0yt) [reply scripts/cmake/z_vcpkg_fixup_rpath.cmake:13]: I deliberately decided against using that pattern.

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by JavierMatosD** [comment]
@ras0219, @data-queue, @vicroms, @BillyONeal, and I agree that fixing regex handling is absolutely good. However, expanding the definition of where debug tools should go depends on the specific use cases and we don't know of any generic use cases at this time. 

That said, the DBUS situation specifically is unusual and we do believe we need to deploy the debug Daemon exe.

**Comment by dg0yt** [comment]
> expanding the definition of where debug tools should go depends on the specific use cases and we don't know of any generic use cases at this time.

Sorry, but I cannot follow.
Do you want to say that `tools/<port>/debug` and `manual-tools/<port>/debug` may contain something else than debug tools?
`tools/<port>/debug/bin` is populated by `vcpkg_configure/build/install_make`, Qt5 and tcl, so there is a valid defect if the rpaths aren't fixed.

For numbers, see https://github.com/microsoft/vcpkg/pull/37736#issuecomment-2035415913. Of course, this includes `foo-config` scripts.
For limiting the ongoing expansion, please resolve https://github.com/microsoft/vcpkg/issues/17607.

**Review by BillyONeal — APPROVED** [review]
THANK YOU for the tests.

**Comment by dg0yt** [comment]
Nice, can we move the dbus debug tools to a proper location now?

**Comment by BillyONeal** [comment]
> Nice, can we move the dbus debug tools to a proper location now?

My understanding is that the dbus situation is different because it's about how the library finds the exe but I didn't pay super close attention to the details there; I poked @data-queue and @ras0219-msft who did

**Comment by dg0yt** [comment]
> it's about how the library finds the exe

I don't think I ever saw a mention of this twist. It was always about the exes finding their libs.  And a little bit about cmake prefix and find stuff. Nothing specific about dbus, except the reason why the debug tools would be needed.

**Comment by ras0219-msft** [comment]
> Nothing specific about dbus, except the reason why the debug tools would be needed.

The specific weird thing about dbus is that the executable is launched by the library, which means it's like a configuration-dependent "cacert.pem" thing that needs to be deployed with the final user's application. This means that users of the debug version of libdbus actually do need a debug variant of the executable.

It is a non-goal that the debug variant of dbus-daemon is executable from the installed directory; it needs to be copied by the user into their overall product and probably rpath-adjusted to work in that environment.

**Pull request merged**